packages/compass-components/src/components/workspace-tabs/tab.tsx

```diff
@@ -189,6 +203,14 @@ function Tab({
     props
   );
 
+  const themeClass = useMemo(() => {
+    if (!tabTheme) {
+      return darkMode ? tabDarkThemeStyles : tabLightThemeStyles;
+    }
+
+    return css(tabTheme);
+  }, [tabTheme, darkMode, tabDarkThemeStyles, tabLightThemeStyles, css]);
```

Comment:
```suggestion
  }, [tabTheme, darkMode]);
```
Those were outside of the Component so not needed here as deps.